Core/src/org/sleuthkit/autopsy/healthmonitor/HealthMonitorCaseEventListener.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.healthmonitor;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.sleuthkit.autopsy.casemodule.Case;
+
+/**
+ * Listener for case events
+ */
+final class HealthMonitorCaseEventListener implements PropertyChangeListener {
```

Comment:
It seems to me we can simplify things by making the monitor itself an event listener. We would register the monitor instance as a subscriber in getInstance, not the ServicesHealthMonitor constructor, to avoid leaking the "this" pointer. That would eliminate this class class with its additional thread pool and would allow ServicesHealthMonitor.DatabaseWriteTask to be private.